src/components/Form/index.tsx

```diff
@@ -42,9 +42,9 @@ export interface FormValidationError {
 export interface FormProps<FormValues> extends StandardProps {
   /**
    * Shows the given message if the user wants to navigate
-   * with changes being made.
+   * with changes being made or renders custom component with message if provided.
    */
-  prompt?: string;
+  prompt?: (changed: boolean) => React.ReactChild | string;
```

Comment:
it doesn't have to. It compiles without any problems, and recognizes types properly.

Reply:
[t.ly/yZEXD](url)
```ts
type Foo = (changed: boolean) => {} | string;
type Bar = ((changed: boolean) => {}) | string;

const foo: Foo = ''; // Type '""' is not assignable to type 'Foo'.(2322)
const bar: Bar = '';
```

I guess Ivan's point is right. It will compile well for sure, but you won't be able to pass a string as a value of `prompt` using TS.

Reply:
I see what you mean now, will fix it. Thanks @IKolosynskyi and @dimabory

Reply:
updated